matcher.go

```diff
@@ -1,33 +1,31 @@
 package errors
 
+// Matcher match a given error
 type Matcher interface {
-	Match(required ...*Kind) bool
+	// Is return true if the err match
+	Is(err error) bool
 }
 
+// Is check if err match all matchers
 func Is(err error, matchers ...Matcher) bool {
-	if err == nil {
-		return false
-	}
-
-	e, ok := err.(*Error)
-	if !ok {
-		return false
+	var are int
```

Comment:
```go
func Is(err error, matchers ...Matcher) bool {
	for _, m := range matchers {
		if !m.Is(err) {
			return false
		}
	}
	return true
}
```

This implementation exists faster if a matcher that is not the last is not fulfilled.

Also, the result is `true` when no matchers are passed. Is this behaviour something we want?
IMHO, it should be false when there are no matchers because, by definition, A cannot be any of B if there is no B.

Implementation with that logic:

```go
func Is(err error, matchers ...Matcher) bool {
	if len(matchers) == 0 {
		return false
	}

	for _, m := range matchers {
		if !m.Is(err) {
			return false
		}
	}
	return true
}
```